travis/self_tests

```diff
@@ -1,77 +1,406 @@
+            repo_path = os.path.join(self.git_work_dir, '.git')
+            branch_name = GitRun(repo_path).get_branch_name()
+            version = (branch_name.replace('_', '-').split('-')[:1]
+                       if branch_name else False)
+            version = version[0] if version else None
+        if not version:
+            print travis_helpers.yellow(
+                'Undefined environment variable'
+                ' `VERSION`.\nSet `VERSION` for '
+                'compatibility with guidelines by version.')
+        return version
+
+    def pylint_run(self, is_pr, version, expected_errors, errors_dict):
+        odoo_version = self.version_validate(version)
+        extra_params_cmd = self.get_extra_params(odoo_version)
+        extra_info = "extra_params_cmd %s " % extra_params_cmd
+        print extra_info
+        conf = ["--rcfile=%s" % self.pylint_rcfile]
+        cmd = conf + extra_params_cmd + self.modules_cmd
+        real_errors = self.pylint(cmd)
+        count_errors = sum(real_errors.values())
+        count_info = "count_errors %s" % count_errors
+        print count_info
+        if is_pr:
+            modules_changed = get_modules_changed(
+                self.git_work_dir, self.branch_base)
+            modules_changed_cmd = []
+            for module_changed in modules_changed:
+                modules_changed_cmd.extend([module_changed])
+            subpaths = run_pylint.get_subpaths(list(modules_changed_cmd))
+
+            conf = ["--rcfile=%s" % self.pylint_rcfile_pr]
+            cmd = conf + extra_params_cmd + subpaths
+            pr_real_errors = self.pylint(cmd)
+            if pr_real_errors:
+                pr_errors = sum(pr_real_errors.values())
+                print travis_helpers.yellow(
+                    "Found %s errors in modules changed." % (pr_errors))
+                if pr_errors < 0:
+                    real_errors = pr_real_errors
+                else:
+                    new_dict = {}
+                    for val in real_errors:
+                        new_dict[val] = (
+                            new_dict.get(val, 0) + real_errors[val])
+                    for val in pr_real_errors:
+                        new_dict[val] = (
+                            new_dict.get(val, 0) + pr_real_errors[val])
+                    real_errors = new_dict
+                    count_errors = sum(real_errors.values())
+
+        # expected_errors = int(os.environ.get('PYLINT_EXPECTED_ERRORS', 0))
+        self.assertNotEqual(count_errors, -1,
+                            travis_helpers.yellow('Python modules not found'))
+        self.assertEqual(
+            count_errors, expected_errors,
+            travis_helpers.red("pylint expected errors %s, found %s!" % (
+                expected_errors, count_errors)))
+        # Expected vs found errors
+        self.assertEqual(
+            sorted(real_errors.items()), sorted(errors_dict.items()))
```

Comment:
I think that using directly the dictionary could work fine too
I mean, `self.assertEqual(real_errors, errors_dict)`